x-pack/legacy/plugins/task_manager/task_pool.ts

```diff
@@ -66,50 +81,125 @@ export class TaskPool {
   };
 
   public cancelRunningTasks() {
-    this.logger.debug(`Cancelling running tasks.`);
+    this.logger.debug(
+      i18n.translate('xpack.taskManager.taskPool.cancelRunningTasks', {
+        defaultMessage: 'Cancelling running tasks.',
+      })
+    );
     for (const task of this.running) {
       this.cancelTask(task);
     }
   }
 
-  private async attemptToRun(tasks: TaskRunner[]) {
-    for (const task of tasks) {
-      if (this.availableWorkers > 0) {
-        if (await task.markTaskAsRunning()) {
-          this.running.add(task);
-          task
-            .run()
-            .catch(err => {
-              this.logger.warn(`Task ${task} failed in attempt to run: ${err.message}`);
-            })
-            .then(() => this.running.delete(task));
-        } else {
-          this.logger.warn(`Failed to mark Task ${task} as running`);
-        }
-      } else {
-        return false;
+  private async attemptToRun(tasks: TaskRunner[]): Promise<TaskPoolRunResult> {
+    const [tasksToRun, leftOverTasks] = partitionListByCount(tasks, this.availableWorkers);
+    if (tasksToRun.length) {
+      performance.mark('attemptToRun_start');
+      await Promise.all(
+        tasksToRun.map(async task => {
+          const result = await task
+            .markTaskAsRunning()
+            .then((hasTaskBeenMarkAsRunning: boolean) =>
+              hasTaskBeenMarkAsRunning
+                ? (asOk(task) as Ok<TaskRunner>)
+                : (asErr([task, { message: VERSION_CONFLICT_MESSAGE }]) as Err<[TaskRunner, Error]>)
+            )
+            .catch((ex: Error) => asErr([task, ex]) as Err<[TaskRunner, Error]>);
+
+          this.handleMarkAsRunning(result);
+        })
+      );
+
+      performance.mark('attemptToRun_stop');
+      performance.measure('taskPool.attemptToRun', 'attemptToRun_start', 'attemptToRun_stop');
+    }
+
+    if (leftOverTasks.length) {
+      if (this.availableWorkers) {
+        return this.attemptToRun(leftOverTasks);
       }
+      return TaskPoolRunResult.RanOutOfCapacity;
     }
+    return TaskPoolRunResult.RunningAllClaimedTasks;
+  }
 
-    return true;
+  private handleMarkAsRunning(result: Ok<TaskRunner> | Err<[TaskRunner, Error]>) {
+    either<TaskRunner, [TaskRunner, Error], void>(
```

Comment:
It's not ignored, it's passed into the `either`.. I'll see if I can make it clearer.
To me it's clear because of years of work with Functional constructs, but I'll try and make it more imperative.